specification/chaos/resource-manager/Microsoft.Chaos/preview/2022-10-01-preview/types/experiments.json

```diff
@@ -0,0 +1,812 @@
+          "pattern": "^P(\\d+Y)?(\\d+M)?(\\d+D)?(T(\\d+H)?(\\d+M)?(\\d+(\\.\\d+)?S)?)?$"
+        },
+        "parameters": {
+          "description": "List of key value pairs.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/keyValuePair"
+          },
+          "x-ms-identifiers": [
+            "key"
+          ]
+        },
+        "selectorId": {
+          "type": "string",
+          "description": "String that represents a selector.",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "duration",
+        "parameters",
+        "selectorId"
+      ],
+      "additionalProperties": false,
+      "x-ms-discriminator-value": "continuous"
+    },
+    "keyValuePair": {
+      "description": "A map to describe the settings of an action.",
+      "type": "object",
+      "properties": {
+        "key": {
+          "description": "The name of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        },
+        "value": {
+          "description": "The value of the setting for the action.",
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "key",
+        "value"
+      ],
+      "additionalProperties": false
+    },
+    "selector": {
+      "type": "object",
+      "description": "Model that represents a selector in the Experiment resource.",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Enum of the selector type.",
+          "enum": [
+            "Percent",
+            "Random",
+            "Tag",
+            "List"
+          ],
+          "x-ms-enum": {
+            "name": "SelectorType",
+            "modelAsString": false
+          }
+        },
+        "id": {
+          "type": "string",
+          "description": "String of the selector ID.",
+          "minLength": 1
+        },
+        "targets": {
+          "type": "array",
+          "description": "List of Target references.",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/definitions/targetReference"
+          }
+        },
+        "filter": {
+          "$ref": "#/definitions/filter"
+        }
+      },
+      "required": [
+        "type",
+        "id",
+        "targets"
+      ],
+      "additionalProperties": false
+    },
+    "filter": {
+      "type": "object",
+      "description": "Model that represents available filter types that can be applied to a targets list.",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Enum that discriminates between filter types. Currently only `Simple` type is supported.",
+          "enum": [
+            "Simple"
+          ],
+          "x-ms-enum": {
+            "name": "FilterType",
+            "modelAsString": false
+          }
```

Comment:
Can this enum be removed from the model? I am not 100% sure how autorest will handle the discriminator being an enum, and the allowed values are already defined in `x-ms-discriminator-value`.

Example of the `action` model using only a `string`: https://github.com/Azure/azure-rest-api-specs/blob/8ff0e3b8dc12cd793f4f2208d76f9f3a7f51176c/specification/chaos/resource-manager/Microsoft.Chaos/preview/2022-07-01-preview/types/experiments.json#L184-L202

Reply:
Verified that AutoRest handles discriminator as string (converting enum to string before check).